src/roles.js

```diff
@@ -17,25 +19,51 @@ export const LEFT_ROLE_ID = '8ced989b1904606b'
  */
 
 /**
- * @typedef {object} Capability
+ * @typedef {object} Role
+ * @property {RoleId} roleId
  * @property {string} name
  * @property {Record<import('@mapeo/schema').MapeoDoc['schemaName'], DocCapability>} docs
- * @property {RoleId[]} roleAssignment
+ * @property {RoleIdAssignableToOthers[]} roleAssignment
  * @property {Record<import('./core-manager/core-index.js').Namespace, 'allowed' | 'blocked'>} sync
  */
 
 /**
- * @typedef {typeof COORDINATOR_ROLE_ID | typeof MEMBER_ROLE_ID | typeof BLOCKED_ROLE_ID | typeof LEFT_ROLE_ID} RoleId
+ * @typedef {(
+ *   typeof CREATOR_ROLE_ID |
+ *   typeof COORDINATOR_ROLE_ID |
+ *   typeof MEMBER_ROLE_ID |
+ *   typeof BLOCKED_ROLE_ID |
+ *   typeof LEFT_ROLE_ID |
+ *   typeof NO_ROLE_ID
+ * )} RoleId
  */
 
 /**
- * This is currently the same as 'Coordinator' capabilities, but defined
- * separately because the creator should always have ALL capabilities, but we
- * could edit 'Coordinator' capabilities in the future
+ * @typedef {Extract<RoleId,
+ *   typeof COORDINATOR_ROLE_ID |
+ *   typeof MEMBER_ROLE_ID |
+ *   typeof BLOCKED_ROLE_ID
+ * >} RoleIdAssignableToOthers
+ */
+
+/**
+ * @typedef {Extract<RoleId,
+ *   typeof COORDINATOR_ROLE_ID |
+ *   typeof MEMBER_ROLE_ID |
+ *   typeof BLOCKED_ROLE_ID |
+ *   typeof LEFT_ROLE_ID
+ * >} RoleIdAssignableToAnyone
+ */
```

Comment:
I think better to have both of these as runtime checks to e.g.

```js
const ROLE_ID_ASSIGNABLE_TO_ANYONE = /** @type {const} */ ([
  COORDINATOR_ROLE_ID,
  MEMBER_ROLE_ID,
  BLOCKED_ROLE_ID,
  LEFT_ROLE_ID,
])
/** @typedef {typeof ROLE_ID_ASSIGNABLE_TO_ANYONE} RoleIdAssignableToAnyone */
```

and when used as a type we should also add an assertion.

Reply:
Done in f24c85e3d807ce58e3f4b99a148f6ee1c9d345b6.